### PR TITLE
python312Packages.pymssql: init at 2.3.0

### PR DESCRIPTION
**Pull request by sith-lord-vader**
## Description of changes
Added the ***[pymssql](https://github.com/pymssql/pymssql)*** Python package to nixpkgs.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [x] `sandbox = true`
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/development/python-modules/pymssql/default.nix [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+    tomli
+  ];
+
+  buildInputs = [
+    freetds
+    krb5
+    openssl
+  ];
+
+  dependencies = [ sqlparse ];
```
Comment (dotlambda) [inline pkgs/development/python-modules/pymssql/default.nix:40]: Why?

File: pkgs/development/python-modules/pymssql/default.nix [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+
+  nativeCheckInputs = [
+    gevent
+    psutil
+    pytestCheckHook
+    sqlalchemy
+  ];
+
+  doCheck = true;
```
Comment (dotlambda) [inline pkgs/development/python-modules/pymssql/default.nix:49]:
```suggestion
```

File: pkgs/development/python-modules/pymssql/default.nix (hunk 1 above)
Comment (dotlambda) [reply pkgs/development/python-modules/pymssql/default.nix:40]: No, just force-push.

Comment (sith-lord-vader) [reply pkgs/development/python-modules/pymssql/default.nix]: Got it.

File: pkgs/development/python-modules/pymssql/default.nix [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+
+  build-system = [
+    cython
+    setuptools-scm
+    tomli
+  ];
+
+  buildInputs = [
+    freetds
+    krb5
```
Comment (dotlambda) [inline pkgs/development/python-modules/pymssql/default.nix:35]: This is supposed to be the C library? See https://github.com/NixOS/nixpkgs/pull/333058.

File: pkgs/development/python-modules/pymssql/default.nix [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+    sqlalchemy
+  ];
+
+  pythonImportsCheck = [ "pymssql" ];
+
+  meta = with lib; {
+    changelog = "https://github.com/pymssql/pymssql/blob/v${version}/ChangeLog.rst";
+    description = "Simple database interface for Python that builds on top of FreeTDS to provide a Python DB-API (PEP-249) interface to Microsoft SQL Server";
+    homepage = "https://github.com/pymssql/pymssql";
+    license = licenses.lgpl21;
```
Comment (dotlambda) [inline pkgs/development/python-modules/pymssql/default.nix:52]: `lgp21Only` or `lgpl21Plus`?

File: pkgs/development/python-modules/pymssql/default.nix (hunk 3 above)
Comment (sith-lord-vader) [reply pkgs/development/python-modules/pymssql/default.nix:35]: Yes. I'll also explicitly inherit krb5 from pkgs

Comment (dotlambda) [reply pkgs/development/python-modules/pymssql/default.nix:35]: Please call it `krb5-c` as in that PR.

File: pkgs/top-level/python-packages.nix [hunk 5]
```diff
@@ -11809,6 +11809,8 @@ self: super: with self; {
 
   pymsgbox = callPackage ../development/python-modules/pymsgbox { };
 
+  pymssql = callPackage ../development/python-modules/pymssql { krb5-c = pkgs.krb5; };
```
Comment (dotlambda) [inline pkgs/top-level/python-packages.nix:11812]:
```suggestion
  pymssql = callPackage ../development/python-modules/pymssql {
    krb5-c = pkgs.krb5;
  };
```

**Comment by sith-lord-vader** [comment]
@dotlambda any update on this?

File: pkgs/development/python-modules/pymssql/default.nix [hunk 6]
```diff
@@ -0,0 +1,55 @@
...
+    sqlalchemy
+  ];
+
+  pythonImportsCheck = [ "pymssql" ];
+
+  meta = with lib; {
+    changelog = "https://github.com/pymssql/pymssql/blob/v${version}/ChangeLog.rst";
+    description = "Simple database interface for Python that builds on top of FreeTDS to provide a Python DB-API (PEP-249) interface to Microsoft SQL Server";
+    homepage = "https://github.com/pymssql/pymssql";
+    license = licenses.lgpl21Only;
```
Comment (dotlambda) [inline pkgs/development/python-modules/pymssql/default.nix:52]:
```suggestion
    license = licenses.lgpl21Plus;
```
https://github.com/pymssql/pymssql/blob/v2.3.1/setup.py#L11

**Comment by dotlambda** [comment]
Result of `nixpkgs-review pr 334385` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>4 packages built:</summary>
  <ul>
    <li>python311Packages.pymssql</li>
    <li>python311Packages.pymssql.dist</li>
    <li>python312Packages.pymssql</li>
    <li>python312Packages.pymssql.dist</li>
  </ul>
</details>

**Pull request merged**